### PR TITLE
Allow dependent tx issuance over the wallet API

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

The X-chain mempool post Cortina does not allow dependent transactions to be included. For example:

- Tx A consumes UTXO A and produces UTXO B.
- Tx B consumes UTXO B and produces UTXO C.

The mempool disallows this to avoid mempool DoSes. However, the wallet API is explicitly designed to enable issuance of dependent txs.

## How this works

Honors the mempool requirement of only having no dependent txs while making the API seem like there are dependent txs in the mempool. In the example above, Tx A will be issued into the mempool. If Tx B is then issued while Tx A is still processing, Tx B will be enqueued until Tx A is accepted. After Tx A is accepted, Tx B will be issued into the mempool.

This is a fairly minimal implementation that does not attempt to parallelize independent txs.

## How this was tested

Issued many concurrent txs onto Fuji

### CODE REVIEW

**Review by coffeeavax — APPROVED** [review]

File: vms/avm/wallet_service.go [hunk 1]
```diff
@@ -28,7 +28,36 @@ type WalletService struct {
...
+	if _, ok := w.pendingTxs.Get(txID); !ok {
+		return
+	}
+
+	w.vm.ctx.Log.Info("tx decided over wallet API",
+		zap.Stringer("txID", txID),
+	)
 	w.pendingTxs.Delete(txID)
+
+	for {
```
Comment (gyuho) [inline vms/avm/wallet_service.go:40]: Can we document or just help me understand, why pop off oldest transaction only once, and exit the for-loop on success? Trying to understand how `decided` function relates to this operation.

Comment (gyuho) [reply vms/avm/wallet_service.go:40]: To rephrase my question,

> Allow dependent tx issuance

In what cases, dependent tx failed to be issued, so we need another tx issuance to trigger the tx issue

**Review by patrick-ogrady — APPROVED** [review]

File: vms/avm/wallet_service.go (hunk 1 above)
Comment (StephenButtolph) [reply vms/avm/wallet_service.go:40]: If the user is issuing virtuous txs - verification will never fail. However, I didn't want to just have the node crash (or fall into an infinite loop) in this situation... So instead I dropped the tx and logged a warning.

In all likelihood - if one tx is dropped they all will be.

Comment (gyuho) [reply vms/avm/wallet_service.go:40]: > ... didn't want .. or fall into an infinite loop

Oh makes sense.

**Review by gyuho — APPROVED** [review]
> the wallet API is explicitly designed to enable issuance of dependent txs

Now makes sense. Thanks for the context.

**Pull request merged**